src/components/timeline/TrackThread.js

```diff
@@ -94,8 +98,18 @@ class TimelineTrackThread extends PureComponent<Props> {
    * This will select the leaf-most stack frame or call node.
    */
   _onSampleClick = (sampleIndex: IndexIntoSamplesTable) => {
-    const { threadIndex, selectLeafCallNode, focusCallTree } = this.props;
-    selectLeafCallNode(threadIndex, sampleIndex);
+    const {
+      threadIndex,
+      selectLeafCallNode,
+      selectRootCallNode,
+      focusCallTree,
+      invertCallstack,
+    } = this.props;
+    if (invertCallstack) {
+      selectRootCallNode(threadIndex, sampleIndex);
+    } else {
+      selectLeafCallNode(threadIndex, sampleIndex);
+    }
```

Comment:
Maybe add a comment explaining this condition a bit more. For example `When we're displaying the inverted call stack, the "leaf" call node we're interesting in is actually displayed as the "root" of the tree.`.

I wonder if this condition shouldn't be in `selectLeafCallNode` actually. I think the `TrackThread` component shouldn't bother about the fact we're in invertCallStack mode... but I'm not sure so let's keep it like you did here, we can anyway revisit later if we need it.

Reply:
I see your point about where to keep that condition now.